File: front/src/service/ProductService.js
import axios from 'axios';

const PRODUCT_API_URL = "http://localhost:8080//api/products" // back->src->controller:@RequestMapping("/api/products") 找到這一個
class ProductService{
    getProducts(){
        return axios.get(PRODUCT_API_URL);
    }
    getProductById(productId){
        return axios.get(`${PRODUCT_API_URL}/${productId}`);
    }
    deleteProduct(productId){
        return axios.delete(`${PRODUCT_API_URL}/${productId}`);
    }
    createproduct(product){
        return axios.post(PRODUCT_API_URL, product);
    }
    updateProduct(product, productId){
        return axios.put(`${PRODUCT_API_URL}/${productId}`, product);
    }
}

export default new ProductService();

File: back/src/main/java/com/example/back/service/ProductService.java
package com.example.back.service;

import com.example.back.model.Product;
import java.util.List;
import java.util.Optional;

//宣告函數(空值)
public interface ProductService{
    List<Product> getAllProducts();
    Optional<Product> getProductById(Long id);
    Product saveProduct(Product product); //儲存紀錄
    Product updateProduct(Product product); //更新紀錄
    void deleteProduct(Long id);//刪除紀錄
}